### PR TITLE
Fix compatibility

### PR DESCRIPTION
**Pull request by cvium**

### CODE REVIEW

File: Emby.Kodi.SyncQueue/API/SyncAPI.cs [hunk 1]
```diff
@@ -100,14 +99,14 @@ public SyncUpdateInfo Get(GetLibraryItems request)
...
-            _logger.Debug("Emby.Kodi.SyncQueue:  Request processed... Returning result...");
+            _logger.LogDebug("Emby.Kodi.SyncQueue:  Request processed... Returning result...");
             return x.Result;
         }
 
         public SyncUpdateInfo Get(GetLibraryItemsQuery request)
         {
-            _logger.Info(String.Format("Emby.Kodi.SyncQueue:  Sync Requested for UserID: '{0}' with LastUpdateDT: '{1}'", request.UserID, request.LastUpdateDT));
-            _logger.Debug("Emby.Kodi.SyncQueue:  Processing message...");
+            _logger.LogInformation(String.Format("Emby.Kodi.SyncQueue:  Sync Requested for UserID: '{0}' with LastUpdateDT: '{1}'", request.UserID, request.LastUpdateDT));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/API/SyncAPI.cs]: Same

File: Emby.Kodi.SyncQueue/API/SyncAPI.cs [hunk 2]
```diff
@@ -194,25 +195,26 @@ public SyncUpdateInfo Get(GetLibraryItemsQuery request)
                     }
                 });
 
-                result = items.SelectMany(i => ApiUserCheck.TranslatePhysicalItemToUserLibrary(i, user, _libraryManager)).Select(i => i.GetClientId()).Distinct().ToList();
+                result = items.SelectMany(i => ApiUserCheck.TranslatePhysicalItemToUserLibrary(i, user, _libraryManager)).Select(i => i.Id.ToString("N")).Distinct().ToList();
                 
                 if (result.Count > 0)
                 {
-                    _logger.Info(String.Format("Emby.Kodi.SyncQueue:  Added Items Found: {0}", string.Join(",", result.ToArray())));
+                    _logger.LogInformation(String.Format("Emby.Kodi.SyncQueue:  Added Items Found: {0}", string.Join(",", result.ToArray())));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/API/SyncAPI.cs]: The `.ToArray()` isn't needed.

File: Emby.Kodi.SyncQueue/API/SyncAPI.cs [hunk 3]
```diff
@@ -226,33 +228,35 @@ public SyncUpdateInfo Get(GetLibraryItemsQuery request)
                 //    }
                 //});
 
-                if (data != null && data.Count > 0)
+                if (data != null && data.Count() > 0)
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/API/SyncAPI.cs]: Why `.Count()` and not `.Count` `data` is a list, right?

File: Emby.Kodi.SyncQueue/API/SyncAPI.cs [hunk 4]
```diff
@@ -226,33 +228,35 @@ public SyncUpdateInfo Get(GetLibraryItemsQuery request)
...
                     });
                 }
 
-                //result = items.SelectMany(i => ApiUserCheck.TranslatePhysicalItemToUserLibrary(i, user, _libraryManager, true)).Select(i => i.GetClientId()).Distinct().ToList();
+                //result = items.SelectMany(i => ApiUserCheck.TranslatePhysicalItemToUserLibrary(i, user, _libraryManager, true)).Select(i => i.Id.ToString("N")).Distinct().ToList();
 
                 if (result.Count > 0)
                 {
-                    _logger.Info(String.Format("Emby.Kodi.SyncQueue:  Removed Items Found: {0}", string.Join(",", result.ToArray())));
+                    _logger.LogInformation(String.Format("Emby.Kodi.SyncQueue:  Removed Items Found: {0}", string.Join(",", result.ToArray())));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/API/SyncAPI.cs]: Drop the format and `ToArray()`

File: Emby.Kodi.SyncQueue/API/SyncAPI.cs [hunk 5]
```diff
@@ -265,25 +269,25 @@ public SyncUpdateInfo Get(GetLibraryItemsQuery request)
...
                 });
 
-                result = items.SelectMany(i => ApiUserCheck.TranslatePhysicalItemToUserLibrary(i, user, _libraryManager)).Select(i => i.GetClientId()).Distinct().ToList();
+                result = items.SelectMany(i => ApiUserCheck.TranslatePhysicalItemToUserLibrary(i, user, _libraryManager)).Select(i => i.Id.ToString("N")).Distinct().ToList();
 
 
                 if (result.Count > 0)
                 {
-                    _logger.Info(String.Format("Emby.Kodi.SyncQueue:  Updated Items Found: {0}", string.Join(",", result.ToArray())));
+                    _logger.LogInformation(String.Format("Emby.Kodi.SyncQueue:  Updated Items Found: {0}", string.Join(",", result.ToArray())));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/API/SyncAPI.cs]: Same

File: Emby.Kodi.SyncQueue/Data/DbRepo.cs [hunk 6]
```diff
@@ -272,55 +270,55 @@ public void WriteLibrarySync(List<LibItem> Items, int status, CancellationToken
                     }
                     else
                     {
-                        logger.Debug(String.Format("Emby.Kodi.SyncQueue: NewTime: {0}  OldTime: {1}   Status: {2}", newTime, rec.LastModified, status));
+                        logger.LogDebug(String.Format("Emby.Kodi.SyncQueue: NewTime: {0}  OldTime: {1}   Status: {2}", newTime, rec.LastModified, status));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/Data/DbRepo.cs]: Same

File: Emby.Kodi.SyncQueue/Data/DbRepo.cs [hunk 7]
```diff
@@ -272,55 +270,55 @@ public void WriteLibrarySync(List<LibItem> Items, int status, CancellationToken
...
                         newRec = null;
                     }
 
                     if (newRec != null)
                     {
-                        logger.Debug(String.Format("Emby.Kodi.SyncQueue:  {0} ItemId: '{1}'", statusType, newRec.ItemId));
+                        logger.LogDebug(String.Format("Emby.Kodi.SyncQueue:  {0} ItemId: '{1}'", statusType, newRec.ItemId.ToString("N")));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/Data/DbRepo.cs]: Same

File: Emby.Kodi.SyncQueue/Data/DbRepo.cs [hunk 8]
```diff
@@ -272,55 +270,55 @@ public void WriteLibrarySync(List<LibItem> Items, int status, CancellationToken
...
                     }
                     else
                     {
-                        logger.Debug(String.Format("Emby.Kodi.SyncQueue:  ItemId: '{0}' Skipped", i.Id));
+                        logger.LogDebug(String.Format("Emby.Kodi.SyncQueue:  ItemId: '{0}' Skipped", i.Id.ToString("N")));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/Data/DbRepo.cs]: Same

File: Emby.Kodi.SyncQueue/Data/DbRepo.cs [hunk 9]
```diff
@@ -272,55 +270,55 @@ public void WriteLibrarySync(List<LibItem> Items, int status, CancellationToken
...
                     }
                 }
 
                 if (newRecs.Count > 0)
                 {
 
-                    logger.Debug(String.Format("Emby.Kodi.SyncQueue: {0}", json.SerializeToString(newRecs)));
+                    logger.LogDebug(String.Format("Emby.Kodi.SyncQueue: {0}", json.SerializeToString(newRecs)));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/Data/DbRepo.cs]: Same

File: Emby.Kodi.SyncQueue/Data/DbRepo.cs [hunk 10]
```diff
@@ -272,55 +270,55 @@ public void WriteLibrarySync(List<LibItem> Items, int status, CancellationToken
...
+                    logger.LogDebug("THIS IS AFTER LINQ STARTING COMMIT!");
                     itemRecs.Commit(data);
-                    logger.Debug(String.Format("Emby.Kodi.SyncQueue: {0}", json.SerializeToString(data)));
-                    logger.Debug("THIS IS AFTER LINQ FINISHED COMMIT!");
+                    logger.LogDebug(String.Format("Emby.Kodi.SyncQueue: {0}", json.SerializeToString(data)));
+                    logger.LogDebug("THIS IS AFTER LINQ FINISHED COMMIT!");
 
                     data = itemRecs.Select();
-                    logger.Debug(String.Format("Emby.Kodi.SyncQueue: {0}", json.SerializeToString(data)));                    
+                    logger.LogDebug(String.Format("Emby.Kodi.SyncQueue: {0}", json.SerializeToString(data)));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/Data/DbRepo.cs]: Serilog can print as json, `{@Data}`

File: Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 11]
```diff
@@ -119,11 +119,11 @@ void libraryManager_ItemAdded(object sender, ItemChangeEventArgs e)
...
                 var item = new LibItem()
                 {
-                    Id = e.Item.GetClientId(),
-                    SyncApiModified = (long)(DateTimeOffset.UtcNow.Subtract(new DateTimeOffset(1970, 1, 1, 0, 0, 0, 0, TimeSpan.Zero)).TotalSeconds),
+                    Id = e.Item.Id,
+                    SyncApiModified = (long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds),
                     ItemType = type,
                 };
-                _logger.Debug(string.Format("Emby.Kodi.SyncQueue: ItemAdded added for DB Saving {0}", e.Item.Id));
+                _logger.LogDebug(string.Format("Emby.Kodi.SyncQueue: ItemAdded added for DB Saving {0}", e.Item.Id));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs]: Same

File: Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 12]
```diff
@@ -160,12 +160,12 @@ void libraryManager_ItemUpdated(object sender, ItemChangeEventArgs e)
...
                 {
-                    Id = e.Item.GetClientId(),
-                    SyncApiModified = (long)(DateTimeOffset.UtcNow.Subtract(new DateTimeOffset(1970, 1, 1, 0, 0, 0, 0,TimeSpan.Zero)).TotalSeconds),
+                    Id = e.Item.Id,
+                    SyncApiModified = (long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds),
                     ItemType = type,
                 };
 
-                _logger.Debug(string.Format("Emby.Kodi.SyncQueue: ItemUpdated added for DB Saving {0}", e.Item.Id));
+                _logger.LogDebug(string.Format("Emby.Kodi.SyncQueue: ItemUpdated added for DB Saving {0}", e.Item.Id));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs]: Same

File: Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 13]
```diff
@@ -213,12 +213,12 @@ void libraryManager_ItemRemoved(object sender, ItemChangeEventArgs e)
...
                 {
-                    Id = e.Item.GetClientId(),
-                    SyncApiModified = (long)(DateTimeOffset.UtcNow.Subtract(new DateTimeOffset(1970, 1, 1, 0, 0, 0, 0, TimeSpan.Zero)).TotalSeconds),
+                    Id = e.Item.Id,
+                    SyncApiModified = (long)(DateTime.UtcNow.Subtract(new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc)).TotalSeconds),
                     ItemType = type
                 };
 
-                _logger.Debug(string.Format("Emby.Kodi.SyncQueue: ItemRemoved added for DB Saving {0}", e.Item.Id));
+                _logger.LogDebug(string.Format("Emby.Kodi.SyncQueue: ItemRemoved added for DB Saving {0}", e.Item.Id));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs]: Same

File: Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 14]
```diff
@@ -304,8 +303,8 @@ public Task UpdateLibrary(List<LibItem> Items, string tableName, int status, Can
             
                 DbRepo.Instance.WriteLibrarySync(Items, status, cancellationToken);
 
-                _logger.Info(String.Format("Emby.Kodi.SyncQueue: \"LIBRARYSYNC\" {0} {1} items:  {2}", statusType, Items.Count(),
-                    String.Join(",", Items.Select(i => i.Id).ToArray())));
+                _logger.LogInformation(String.Format("Emby.Kodi.SyncQueue: \"LIBRARYSYNC\" {0} {1} items:  {2}", statusType, Items.Count(),
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs]: Same

File: Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 15]
```diff
@@ -379,7 +378,7 @@ private bool FilterItem(BaseItem item, out int type)
                     break;
                 default:
                     type = -1;
-                    _logger.Debug(String.Format("Emby.Kodi.SyncQueue:  Ingoring Type {0}", typeName));
+                    _logger.LogDebug(String.Format("Emby.Kodi.SyncQueue:  Ingoring Type {0}", typeName));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs]: Same

File: Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs [hunk 16]
```diff
@@ -457,7 +456,7 @@ private bool FilterRemovedItem(BaseItem item, out int type)
...
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/EntryPoints/LibrarySyncNotification.cs]: Same

File: Emby.Kodi.SyncQueue/EntryPoints/UserSyncNotification.cs [hunk 17]
```diff
@@ -125,7 +125,7 @@ private bool FilterItem(BaseItem item, out int type)
                     break;
                 default:
                     type = -1;
-                    _logger.Debug(String.Format("Emby.Kodi.SyncQueue:  Ingoring Type {0}", typeName));
+                    _logger.LogDebug(String.Format("Emby.Kodi.SyncQueue:  Ingoring Type {0}", typeName));
```
Comment (Bond-009) [inline Emby.Kodi.SyncQueue/EntryPoints/UserSyncNotification.cs]: Another one, I'll stop here, you can find+ replace the rest :p

**Comment by cvium** [comment]
@JustAMan https://github.com/jellyfin/jellyfin-plugin-kodisyncqueue/pull/1#issuecomment-459953689 also applies to you ;) I can't explain any of the changes except 
https://github.com/jellyfin/jellyfin-plugin-kodisyncqueue/pull/1/commits/61f3d5a11a4d2dea18f2c3919661ff1e14d43821

File: Emby.Kodi.SyncQueue/Data/DbRepo.cs (hunk 10 above)
Comment (cvium) [reply Emby.Kodi.SyncQueue/Data/DbRepo.cs]: Good to know. Thanks

**Pull request merged**